catkin_ws/src/kpr_testing/src/valves.py

```diff
@@ -0,0 +1,163 @@
+VacSens = "DI1"
+Gripper = "DO2"
+
+#Set succes response
+response = True
+
+#dictionary of pins
+info = {VacSens:["-","     -"], Gripper:["-","     -"]}
+
+#needed for the "pressing keys"
+@contextlib.contextmanager
+def raw_mode(file):
+    old_attrs = termios.tcgetattr(file.fileno())
+    new_attrs = old_attrs[:]
+    new_attrs[3] = new_attrs[3] & ~(termios.ECHO | termios.ICANON)
+    try:
+        termios.tcsetattr(file.fileno(), termios.TCSADRAIN, new_attrs)
+        yield
+    finally:
+        termios.tcsetattr(file.fileno(), termios.TCSADRAIN, old_attrs)
+
+def request(req):
+	global response
+	display()
+#Checking if changes wil be made to used pins 
+	if req.fun == 1:
+		pin = 'DO' + str(req.pin)
+	elif req.fun == 4:
+		pin = 'AO' + str(req.pin)
+	else:
+		rospy.logerr("Writing to unused pin")
+		return SetIOResponse(False)
+	if pin not in info:
+		rospy.logerr("Writing to unused pin")
+		return SetIOResponse(False)
+#Change Dictionary "info" with correct data
+	info[pin][0] = req.state
+	if req.fun == 1:
+		info[pin][1] = "Digital Out"
+	else:
+		info[pin][1] = "Analog Out"
+
+	display()
+	res = response	
+	return SetIOResponse(res)
+		
+def get_message():
+	msg = IOStates()
+	for pin in info:
+		if info[pin][1] == "Digital Out":
+			msg.digital_out_states.append(Digital(int(pin[2]), bool(info[pin][0])))
+		elif info[pin][1] == "Analog Out":
+			msg.analog_out_states.append(Analog(int(pin[2]), float(info[pin][0])))
+		elif info[pin][1] == "Digital In":
+			msg.digital_in_states.append(Digital(int(pin[2]), bool(info[pin][0])))
+		elif info[pin][1] == "Analog In":
+			msg.analog_in_states.append(Analog(int(pin[2]), float(info[pin][0])))
+	return msg
+
+def processKey(key, pub):
+	global response
+	if key == chr(116): #t
+		response = not response
+		display()
+		print "Your response is now: ", response
+		return
+	if key == chr(112): #p
+		pub.publish(get_message())
+		display()
+		print "you have published the data"
+		return
+	if key == chr(105): #i
+		display()
+		print "Changing the input settings"
+		income()
+		return
+	rospy.logwarn('unknown function %s', key)
+
+def income():
+	try:
+		fun = int(raw_input("What is the fun? "))
+		if fun != 5 and fun != 6:
+			display()
+			rospy.logerr("Fun nr not available")
+			return SetIOResponse(False)
+		print fun
+	except ValueError:
+		rospy.logerr("Not a Number")
+		return SetIOResponse(False)
+
+	pinQ = str(raw_input("Which pin you want to set? " ))
+	print pinQ
+
+	if fun == 5:
+		pin = 'DI' + pinQ
+	elif fun == 6:
+		pin = 'AI' + pinQ
+	else:
+		rospy.logerr("Writing to unused pin")
+		return SetIOResponse(False)
+#Checks if pin in dictionary
+	if pin not in info:
+		rospy.logerr("Writing to unused pin")
+		return SetIOResponse(False)
+	print "writing to pin ", pin
+	state = float(raw_input("What is the state? "))
+#Send state to dictionary 
+	info[pin][0] = state 
+#Send fun to dictionary
+	if fun == 5: #fun
+		info[pin][1] = "Digital In"
+	else:
+		info[pin][1] = "Analog In"
```

Comment:
It looks cleaner when you write the new pin in one statement to the new values using `info[pin] = (state, fun)` instead of writing each part separate (but this is taste related, so could be ignored)